src/jit/morph.cpp

```diff
@@ -2017,6 +2017,7 @@ GenTreePtr    Compiler::fgMakeTmpArgNode(unsigned tmpVarNum
         else
         {
             arg = gtNewOperNode(GT_ADDR, TYP_STRUCT, arg);
+            varDsc->lvAddrExposed = 1;
```

Comment:
Instead of directly accessing lvAddrExposed we instead should always call  Compiler::lvaSetVarAddrExposed